### PR TITLE
🪶 fix: Prevent Soft Default Model Spec from Overriding User Selections

### PR DESCRIPTION
**Pull request by danny-avila**
## Summary

Follow-up to #13554: `softDefault` was functioning like a hard `default`. Auto-applying the soft default once persisted it through the conversation pipeline (`lastSelectedSpec`, `lastSelectedModel`, `lastConversationSetup_0.spec`) — the exact keys `hasStoredModelSelection()` checks — so from the second page load on it returned as the ungated `last` selection and outranked previously selected agents and ephemeral endpoint → model selections, every time.

- Treat selection state matching what auto-applying the soft default writes as residue of the soft default itself, not evidence of a user choice — `hasStoredModelSelection()` now ignores `lastSelectedSpec`, `lastSelectedModel`, and agent/assistant entries equal to the soft spec's name and preset.
- Return the soft default via the `softDefault` path instead of `last` when the stored spec is the soft default spec, keeping it re-evaluated and overridable on every load, including under `prioritize: true`.
- Yield to genuine selections: a stored agent, assistant, ephemeral endpoint → model pick, or different spec immediately outranks the soft default.
- Apply the soft default as the fallback default — regardless of stored selections — when the selector offers no ephemeral endpoint → model options (`modelSelect` disabled, or `addedEndpoints`/the endpoints config leaving only agents/assistants), since there is no regular selection to defer to in those deployments.
- Pass the endpoints config into `getDefaultModelSpec` from `ChatRoute` and `useNewConvo` to support the detection above.
- Extend the `getDefaultModelSpec` unit tests from 6 to 20 cases covering auto-application residue, agent/ephemeral/spec selections made after the soft default applied, agent-pointing soft specs, and the no-ephemeral-options edge cases.
- Add `e2e/specs/mock/soft-default.spec.ts` covering the reported flows end to end: fresh instance (cleared localStorage) applies the soft default and keeps it across reloads; a previously selected agent survives reload and new chat; a previous ephemeral endpoint → model selection survives reload and new chat.
- Add a `Mock Provider C` endpoint and `addedEndpoints` to the e2e config so the mock harness exposes a real endpoints menu (agents + endpoint → model) without colliding with existing spec labels.

## Change Type

- [x] Bug fix (non-breaking change which fixes an issue)

## Testing

- `cd client && npx jest src/utils/__tests__/getDefaultModelSpec.test.ts` — 20/20 passing (previously 6).
- `cd client && npx jest src/utils src/hooks src/routes` — all suites passing.
- Full mock e2e suite — 35/35 passing, including the 3 new soft-default tests and every existing spec against the updated e2e config.
- Manual repro of the report: configure a spec with `softDefault: true`, load fresh (spec selected), select an agent or an endpoint + model, then reload or start a new chat — the selection now persists instead of the spec reapplying.

### **Test Configuration**:

- Mock harness, no real keys: `E2E_BASE_URL=http://localhost:3334 MOCK_LLM_PORT=8899 npm run e2e:mock`

## Checklist

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented in any complex areas of my code
- [x] My changes do not introduce new warnings
- [x] I have written tests demonstrating that my changes are effective or that my feature works
- [x] Local unit tests pass with my changes

### CODE REVIEW

**Comment by danny-avila** [comment]
@codex review

File: client/src/utils/endpoints.ts [hunk 1]
```diff
@@ -203,29 +208,69 @@ export function hasModelSelection(selection?: Partial<StoredModelSelection> | nu
...
+    return true;
+  }
+  const included = new Set(addedEndpoints ?? []);
+  return Object.entries(endpointsConfig).some(
+    ([endpoint, config]) =>
+      config != null &&
+      !isAgentsEndpoint(endpoint) &&
+      !isAssistantsEndpoint(endpoint) &&
+      (included.size === 0 || included.has(endpoint)),
+  );
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply client/src/utils/endpoints.ts:273]: Addressed in 31bd2222ea — an allow-list containing only agents/assistants now short-circuits to "no ephemeral options" before the endpoints config loads, so the soft default no longer flips from gated to forced mid-load. Added a unit test covering the not-yet-loaded window.

File: client/src/utils/endpoints.ts [hunk 2]
```diff
@@ -358,34 +403,56 @@ export function applyModelSpecEphemeralAgent({
...
+    return hasStoredModelSelection(softDefaultSpec) ? undefined : { softDefault: softDefaultSpec };
+  };
   if (prioritize === true || !interfaceConfig?.modelSelect) {
     const lastSelectedSpecName = localStorage.getItem(LocalStorageKeys.LAST_SPEC);
     const lastSelectedSpec = list?.find((spec) => spec.name === lastSelectedSpecName);
     if (defaultSpec) {
       return { default: defaultSpec };
     }
+    if (lastSelectedSpec && lastSelectedSpec.name === softDefaultSpec?.name) {
+      return resolveSoftDefault();
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply client/src/utils/endpoints.ts:457]: Good catch — addressed in 253f4f25a6. `resolveSoftDefault()` now short-circuits when the stored conversation's spec is the soft default spec (it is the current selection, however it was reached), so an explicit re-selection of the soft spec survives reloads and new chats instead of losing to older `lastSelectedModel` history. This unifies the prioritized branch with the modelSelect branch's existing behavior. Added unit tests for both branches covering the older-history scenario.

**Comment by danny-avila** [comment]
@codex review

File: client/src/utils/endpoints.ts [hunk 3]
```diff
@@ -160,30 +160,35 @@ function parseStoredModelSelection(
...
   if (!storedModelValue) {
     return false;
   }
 
   try {
     const storedModels = JSON.parse(storedModelValue) as Record<string, string | null | undefined>;
-    return Object.values(storedModels).some(hasSelectionValue);
+    return Object.entries(storedModels).some(
+      ([endpoint, model]) =>
+        hasSelectionValue(model) &&
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply client/src/utils/endpoints.ts]: Addressed in a2430ff085 — residue detection is now limited to spec-named state: only `lastSelectedSpec` or `lastConversationSetup.spec` equal to the soft spec's name is treated as auto-application residue. The `lastSelectedModel`/agent/assistant exemptions are reverted, so a manual selection that merely matches the soft preset counts as a user choice and gates the soft default. The pure auto-apply state is still handled by the current-selection short-circuit on `lastConversationSetup.spec`. Added a regression test for the matching stored-model case and flipped the matching-agent test to expect yielding.

File: e2e/config/librechat.e2e.yaml [hunk 4]
```diff
@@ -26,9 +26,26 @@ endpoints:
...
+      titleConvo: false
+      modelDisplayLabel: 'Mock Provider C'
+
 modelSpecs:
   prioritize: true
   enforce: true
+  # Surfaces the endpoints menu (modelSelect defaults on when addedEndpoints is
+  # set) limited to entries that don't collide with the spec labels above.
+  addedEndpoints:
+    - 'Mock Provider C'
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply e2e/config/librechat.e2e.yaml:49]: Addressed in a2430ff085 — set `enforce: false` in the e2e config with a comment explaining why: `addedEndpoints` deliberately exposes non-spec send paths (Mock Provider C, agents), and `buildEndpointOption` would reject those sends under enforcement. No existing spec relies on enforcement behavior; full mock suite still passes 35/35.

**Comment by danny-avila** [comment]
@codex review
